Formula/m/monocle.rb

```diff
@@ -0,0 +1,17 @@
+class Monocle < Formula
+  desc "See through all BGP data with a monocle"
+  homepage "https://github.com/bgpkit/monocle"
+  url "https://github.com/bgpkit/monocle/archive/refs/tags/v0.7.0.tar.gz"
+  sha256 "2ef5030b906a567b6fba2bc4aa593b8cfaa7e64c28557adc2ced60c89ff2a95d"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: ".")
+  end
+
+  test do
+    system bin/"monocle", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate teadd-gurkath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the help. My apologies for missing the functional test. I've updated the test block to execute one subcommand and match the output with expected value:
```ruby 
  test do
    output = shell_output("#{bin}/monocle time 1735322400 --simple")
    assert_match "2024-12-27T18:00:00+00:00", output
  end
```

Reply:
looks good 👍